[panw] Update readme content (#14271)
elastic/integrations: packages/panw/_dev/build/docs/README.md

# Palo Alto Network Integration for Elastic

## Overview

The Palo Alto Network Integration for Elastic enables collection of logs from Palo Alto Networks' PAN-OS firewalls. This integration facilitates real-time visibility into network
activity, threat detection and security operations.

### Compatibility

This integration is compatible with PAN-OS versions 8.1 through 11.0, although earlier versions may have limited support. It has been tested with logs generated by devices running
PAN-OS versions 7.1 to 11.0.

Support for specific log types varies by PAN-OS version. GlobalProtect logs are supported starting with PAN-OS version 9.1.3. User-ID logs are supported for PAN-OS version 8.1 and
above, while Tunnel Inspection logs are supported for version 9.1 and later.

> This integration can receive logs from syslog via TCP or UDP, or read from log files.

Reviewer: ```suggestion
The Palo Alto Network integration can receive logs from syslog via TCP or UDP, or read from log files.
```

> ## What data does this collect?

Reviewer: ```suggestion
## What data does this integration collect?
```

> This integration collects log messages of the following types:

Reviewer: ```suggestion
The Palo Alto Network integration collects log messages of the following types:
```

* [Correlated Events](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/correlated-events-log-fields)
* [Decryption](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/decryption-log-fields)
* [GTP](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/gtp-log-fields)
* [IP-Tag](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/ip-tag-log-fields)
* [SCTP](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/sctp-log-fields)
* [System](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/system-log-fields)
* [Tunnel Inspection](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/tunnel-inspection-log-fields).

> ### Supported uses

Reviewer: Supported use cases?

Reviewer 2: @mjwolf 
Don't know if this is the right place inside this doc, but can we add somewhere where it makes sense the API calls that we use agains PANW from the integration?

Sure it would need to be updated just in case an API call would change in a future PR or added, but that's probably more a question of a runbook issue.

cc @jamiehynds

Integrating Palo Alto Networks (PANW) with the Elastic Stack creates a powerful solution for transforming raw firewall logs into actionable intelligence, dramatically enhancing
security and operational visibility. This synergy enables advanced use cases including real-time threat detection and hunting through Elastic SIEM, deep network traffic analysis
with intuitive Kibana dashboards, and automated incident response by connecting with Cortex XSOAR. By centralizing and analyzing PANW data, organizations can strengthen their
security posture, optimize network performance, and build a solid data foundation for implementing a Zero Trust architecture.

## What do I need to use this integration?

> ### Prerequisites

Reviewer: H3 Prerequisites might be redundant here. Also in some integrations pages we use Requirements.

Elastic Agent must be installed. For more details, check the Elastic Agent [installation instructions](docs-content://reference/fleet/install-elastic-agents.md). You can install only one Elastic Agent per host.

Elastic Agent is required to stream data from the syslog or log file receiver and ship the data to Elastic, where the events will then be processed via the integration's ingest pipelines.

> Permissions are required to configure syslog monitoring or log forwarding on the firewall appliance.

Reviewer 3: I think nothing else, it's contained within the panw link below. The only other thing to consider is the agent performance settings but that could be covered in the scalability section

## How do I deploy this integration?

### Collect logs via syslog

> To configure syslog monitoring, please follow the steps mentioned in the [_Configure Syslog Monitoring_](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/configure-syslog-monitoring).

Reviewer: ```suggestion
To configure syslog monitoring, follow the steps described in the [Configure Syslog Monitoring](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/use-syslog-for-monitoring/configure-syslog-monitoring) documentation.
```

### Collect logs via log file

> To configure log file monitoring, please follow the steps mentioned in [_Configure Log Forwarding_](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/configure-log-forwarding).

Reviewer: ```suggestion
To configure log file monitoring, follow the steps described in the [Configure Log Forwarding](https://docs.paloaltonetworks.com/pan-os/10-2/pan-os-admin/monitoring/configure-log-forwarding) documentation.
```

### Enable the integration in Elastic

1. In Kibana navigate to **Management** > **Integrations**.
2. In the search bar, type **Palo Alto Next-Gen Firewall**.
3. Select the **Palo Alto Next-Gen Firewall** integration and add it.
4. If needed, install Elastic Agent on the systems which receive syslog messages or log files.
5. Enable and configure only the collection methods which you will use.

>     * **To collect logs via syslog over TCP**, enter the syslog server host and port details.

Author: The manifest file already has required/optional for fields, so this will be shown when configuring the integration (although required fields have no label shown, only optional fields are marked). 

I've also now given all fields descriptions in the manifest, some where missing before and reworded the instuctions here to say which are required.

I think it's better to keep most of the field descriptions in the manifest, since that's where people will be looking when setting up the integration. But for other integrations with fields that need more than a few sentences to explain, I think it would make sense to add to this section.

    * **To collect logs via syslog over UDP**, enter the syslog server host and port details.

    * **To collect logs via log file**, set the file path patterns which will be monitored, in the Paths field.

> 6. Press **Save Integration** to begin collecting logs.

Reviewer: ```suggestion
6. Click **Save Integration** to start collecting logs.
```

### Validate log collection

1. In Kibana, navigate to **Dashboards**.
2. In the search bar, type **Logs PANW**.
3. Select a dashboard overview for the data type you are collecting, and verify the dashboard information is populated.

## Troubleshooting

> For help with Elastic ingest tools, check [Troubleshoot ingestion tools](https://www.elastic.co/docs/troubleshoot/ingest).

Reviewer 3: Also note that error scenarios within Kibana->fleet also point to this common-problems page. The "troubleshooting guide" link points to the link mentioned above
![image](https://github.com/user-attachments/assets/d2dfc8b5-96f9-4792-823e-3405b54af469)

> If events are getting truncated, then increase `max_message_size` option for TCP and UDP input type. It can be found under Advanced Options and can be configured as per requirements.

Reviewer: ```suggestion
If events are truncated, increase `max_message_size` option for TCP and UDP input type. You can find it under Advanced Options and configure it as per requirements.
```

The default value of `max_message_size` is set to 50KiB.

If the TCP input is used, it is recommended that PAN-OS is configured to send syslog messages using the IETF (RFC 5424) format. In addition, RFC 6587 framing (Octet Counting) will
be enabled by default on the TCP input.

> If you want to see the configuration before and after the change(fields `before-change-detail` and `after-change-detail`) in the [config-log](https://docs.paloaltonetworks.com/pan-os/11-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/config-log-fields), please use the following [custom log format in the syslog server profile](https://docs.paloaltonetworks.com/pan-os/11-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/custom-logevent-format):

Reviewer: ```suggestion
To verify the configuration before and after the change (fields `before-change-detail` and `after-change-detail`) in the [config-log](https://docs.paloaltonetworks.com/pan-os/11-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/config-log-fields), use the following [custom log format in the syslog server profile](https://docs.paloaltonetworks.com/pan-os/11-1/pan-os-admin/monitoring/use-syslog-for-monitoring/syslog-field-descriptions/custom-logevent-format):
```

  ``1,$receive_time,$serial,$type,$subtype,2561,$time_generated,$host,$vsys,$cmd,$admin,$client,$result,$path,$before-change-detail,$after-change-detail,$seqno,$actionflags,$dg_hier_level_1,$dg_hier_level_2,$dg_hier_level_3,$dg_hier_level_4,$vsys_name,$device_name,$dg_id,$comment,0,$high_res_timestamp``

## Performance and scaling

> For more information on architectures that can be used for scaling this integration, please see [Ingest Architectures](https://www.elastic.co/docs/manage-data/ingest/ingest-reference-architectures).

Reviewer: ```suggestion
For more information on architectures that can be used for scaling this integration, check the [Ingest Architectures](https://www.elastic.co/docs/manage-data/ingest/ingest-reference-architectures) documentation .
```

## Reference

### ECS field reference

{{fields "panos"}}

### Example event

{{event "panos"}}

### Inputs

> These inputs can be used in this integration

Reviewer: ```suggestion
These inputs can be used in this integration:
```
